### ISSUE TITLE
Default "__tera_one_off" template name used for the string templates creates confusion 

### ISSUE THREAD

**Issue by anzome** [Keats/tera#722]
Tera provides two methods which allows to render the templates without a name and clear posibility to derive a name from somewhere (filename, for example). Though for the code consistency Tera internals consideres name required. That is why default value was introduced.

The issue with this default value is that is requires to be mentioned in docs explicitly. It contains not enough information on its own. Consider something like "NAMELESS STRING TEMPLATE" default name to compare how it could make error messages easier to comprehend even without additional mentions in documentation.

**Comment by anzome** [comment]
I will prepare the PR if the idea got an approve.

**Comment by Keats** [comment]
Yep, this will get changed for the next major version, probably no need to change it now though.

**Comment by anzome** [comment]
Did you changed it already? I can create a PR into separate branch, dedicated for the next major release. Will you plan  to create such branch?

**Comment by Keats** [comment]
The repo will likely be rewritten from scratch

**Comment by Keats** [comment]
Closing as part of the Tera v2 release. If this is still relevant in v2, please reopen — see the 
  migration guide: https://github.com/Keats/tera/blob/master/MIGRATION.md
